ui/imports/shared/views/HistoryView.qml

```diff
@@ -378,4 +387,28 @@ ColumnLayout {
             z: 3
         }
     }
+
+    Connections {
+        target: RootStore
+
+        function onNewDataAvailableChanged() {
+            if (!d.lastRefreshTime || ((Date.now() - d.lastRefreshTime) > (1000 * d.maxSecondsBetweenRefresh))) {
+                d.showRefreshButton = RootStore.newDataAvailable
```

Comment:
> Just curious if there are multiple multiple events received within maxSecondsBetweenRefresh and multiple calls to updateFilter are made. The first call results in susbscribedChainID match with the received chain id and date as well. 

Any match will set `newDataAvailable` to `true` and no match will do nothing.

> Subsequent call does not match some criteria and (If I understand it correctly) newDataAvailable will be set to false in this case, because it was set to false right during updateFilter call.

`newDataAvailable` is set to `false` only when `updateFilter` is called (this is on user request only)

> So `showRefreshButton` will be overwritten from `true` to `false` and it would seem like there are no updates, but in fact there are. Am I right?

I get your point. If the user presses the "new transactions" button `updateFilter` will set the `newDataAvailable` to `false` and if we have an async trigger of events, then `newDataAvailable` as false might come after the `updateFilter` resets the `lastRefreshTime`. This scenario is currently avoided because of the direct connections and `newDataAvailableChange` as `false` comes before the `lastRefreshTime`
I agree. Better to be safe and trigger the timer only if `newDataAvailable` is true.